pkg/persistence/queries.go

```diff
@@ -355,6 +355,41 @@ func checkClaimKeyBan(db queryRower, identifier string) (triesRemaining int, ban
 	return triesRemaining, banDuration, nil
 }
 
+func checkHashId(db *sql.DB, identifier string) (int64, error) {
+	var count int64
+
+	row := db.QueryRow("SELECT COUNT(*) FROM encryption_keys WHERE hash_id = ? and one_time_code IS NULL", identifier)
+	err := row.Scan(&count)
+
+	if err != nil {
+		return 1, err
+	}
+
+	// HashId OTC has been used
+	if count > 0 {
+		return count, err
+	}
+
+	row = db.QueryRow("SELECT COUNT(*) FROM encryption_keys WHERE hash_id = ? AND one_time_code IS NOT NULL", identifier)
+	err = row.Scan(&count)
+
+	if err != nil {
+		return 1, err
+	}
+
+	// HashId OTC exists but has not been used
+	if count > 0 {
+		_, err = db.Exec(`DELETE FROM encryption_keys WHERE hash_id = ? AND one_time_code IS NOT NULL`, identifier)
```

Comment:
I'd probably have done this in a transaction, since this is issued conditionally based on the previous query, but I'm not convinced it matters.

Reply:
Fixed in #57